### PR TITLE
updating & refactoring(existing api calls to transfer them on RTK query)

### PR DESCRIPTION
**Pull request by oleksandrkovaliuk**

### CODE REVIEW

File: src/app/manage/_components/createForm/content/_components/images/images.tsx [hunk 1]
```diff
@@ -47,6 +52,7 @@ export const Images: React.FC<ContentProps> = ({
   selectedAdress,
   handleUpdateFormAndLocalStorage,
 }) => {
+  console.log(images, "images");
```
Comment (o-bilyi) [inline src/app/manage/_components/createForm/content/_components/images/images.tsx]: remove console.log

File: src/app/manage/_components/createForm/content/_components/images/images.tsx [hunk 2]
```diff
@@ -128,16 +134,26 @@ export const Images: React.FC<ContentProps> = ({
       formData.append("user_email", session?.user.email!);
       formData.append("location", selectedAdress!.formattedAddress);
 
-      const res = await UploadListingImages(formData, "form");
+      // const res = await UploadListingImages(formData, "form");
```
Comment (o-bilyi) [inline src/app/manage/_components/createForm/content/_components/images/images.tsx]: remove commented code

File: src/app/manage/_components/createForm/content/_components/images/images.tsx [hunk 3]
```diff
@@ -128,16 +134,26 @@ export const Images: React.FC<ContentProps> = ({
...
 
+      console.log(formData, "d ");
```
Comment (o-bilyi) [inline src/app/manage/_components/createForm/content/_components/images/images.tsx]: remove console.log

File: src/components/_globalDataFetching/index.tsx [hunk 4]
```diff
@@ -2,17 +2,15 @@
...
+// import { getTypeOfPlace } from "@/store/thunks/listings/typeOfPlace";
 
 export const GlobalDataOnLoad = () => {
   const dispath = useDispatch();
   useEffect(() => {
     Promise.all([
       // dispath(getAllCategories() as any),
-      dispath(getTypeOfPlace() as any),
-      dispath(getAllListings() as any),
+      // dispath(getTypeOfPlace() as any),
```
Comment (o-bilyi) [inline src/components/_globalDataFetching/index.tsx]: looks like you can totally remove useEffect and rest code which you used inside useEffect

File: src/store/api/api.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+import { BASE_API_URL } from "../../helpers/constants";
+import { ApiTagsTypes } from "./lib/constants";
+// import { HYDRATE } from "next-redux-wrapper";
+
+export const api = createApi({
+  baseQuery: fetchBaseQuery({
+    baseUrl: BASE_API_URL,
+    prepareHeaders: (headers, { getState }) => {
+      return headers;
```
Comment (o-bilyi) [inline src/store/api/api.ts:11]: try to get a user from the session there

File: src/store/api/endpoints/auth/accesOAuthUser.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        body: {
+          email,
+          user_name,
+          user_lastname,
+          img_url,
+          provider,
+        },
+      }),
+
+      invalidatesTags: ["USER"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/auth/accesOAuthUser.ts]: use enum instead of string

File: src/store/api/endpoints/auth/accesUser.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+const accesUserApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    AccesUser: builder.mutation({
+      query: ({ email, password }: UserTypes) => ({
+        url: apiUrls.auth_acces_user,
+        method: "POST",
+        body: { email, password },
+      }),
+      invalidatesTags: ["USER"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/auth/accesUser.ts]: same as above

File: src/store/api/endpoints/auth/checkAuthType.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+const CheckAuthType = api.injectEndpoints({
+  endpoints: (builder) => ({
+    checkAuthType: builder.mutation({
+      query: ({ email }: { email: string }) => ({
+        url: apiUrls.check_auth_type,
+        method: "POST",
+        body: { email },
+      }),
+      invalidatesTags: ["USER"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/auth/checkAuthType.ts]: same as above

File: src/store/api/endpoints/auth/getUser.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+      query: ({
+        user_name,
+        user_email,
+      }: {
+        user_name: string;
+        user_email: string;
+      }) => ({
+        url: `${apiUrls.getUser}/${user_email}/${user_name}`,
+      }),
+      providesTags: ["CURRENT_USER"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/auth/getUser.ts]: same as above

File: src/store/api/endpoints/listings/getCategories.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+import { apiUrls } from "../../lib/constants";
+import { Category } from "@/store/slices/listingsInfoSlice/type";
+
+const getListingsCategoriesApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    getListingsCategories: builder.query<Category[], void>({
+      query: () => ({
+        url: apiUrls.getListingsCategories,
+      }),
+      providesTags: ["LISTING_CATEGORIES"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/listings/getCategories.ts]: same as above

File: src/store/api/endpoints/listings/getCurrentListing.ts [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+import { api } from "../../api";
+import { apiUrls } from "../../lib/constants";
+
+const getCurrentListingApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    getCurrentListing: builder.query<ListingState, { id: number }>({
+      query: ({ id }: { id: number }) => ({
+        url: `${apiUrls.getCurrentListing}/${id}`,
+      }),
+      providesTags: ["CURRENT_LISTING"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/listings/getCurrentListing.ts]: same as above

File: src/store/api/endpoints/listings/getListings.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+import { api } from "../../api";
+import { apiUrls } from "../../lib/constants";
+
+const getListingsApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    getListings: builder.query<ListingState[], void>({
+      query: () => ({
+        url: apiUrls.getListings,
+      }),
+      providesTags: ["LISTINGS"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/listings/getListings.ts]: same as above

File: src/store/api/endpoints/listings/getTypeOfPlace.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+import { api } from "../../api";
+import { apiUrls } from "../../lib/constants";
+
+const getListingsTypeOfPlaceApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    getListingsTypeOfPlace: builder.query<TypeOfPlace[], void>({
+      query: () => ({
+        url: apiUrls.getListingsTypeOfPlace,
+      }),
+      providesTags: ["LISTING_TYPE_OF_PLACE"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/listings/getTypeOfPlace.ts]: same as above

File: src/store/api/endpoints/listings/getUserListings.ts [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+      query: ({
+        user_name,
+        user_email,
+      }: {
+        user_name: string;
+        user_email: string;
+      }) => ({
+        url: `${apiUrls.getUserListings}/${user_name}/${user_email}`,
+      }),
+      providesTags: ["USER_LISTINGS"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/listings/getUserListings.ts]: same as above

File: src/store/api/endpoints/listings/getVerifiedListings.ts [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+import { apiUrls } from "../../lib/constants";
+import { ListingState } from "../../lib/type";
+
+const getVerifiedListingsApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    getVerifiedListings: builder.query<ListingState[], void>({
+      query: () => ({
+        url: apiUrls.getVerifiedListings,
+      }),
+      providesTags: ["VERIFIED_LISTINGS"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/listings/getVerifiedListings.ts]: same as above

File: src/store/api/endpoints/listings/requestCreateListing.ts [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+          pets_allowed,
+          images,
+          title,
+          aboutplace,
+          placeis,
+          notes,
+          price,
+        },
+      }),
+      invalidatesTags: ["USER_LISTINGS", "VERIFIED_LISTINGS", "LISTINGS"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/listings/requestCreateListing.ts]: same as above

File: src/store/api/endpoints/listings/requestToDeleteListing.ts [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+
+const requestToDeleteListingApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    requestToDeleteListing: builder.mutation<{ id: number }, { id: number }>({
+      query: ({ id }: { id: number }) => ({
+        url: apiUrls.requestToDeleteListing,
+        method: "POST",
+        body: { id },
+      }),
+      invalidatesTags: ["USER_LISTINGS"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/listings/requestToDeleteListing.ts]: same as above

File: src/store/api/endpoints/listings/uploadListingImages.ts [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+
+const uploadListingImagesApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    uploadListingImages: builder.mutation<any, FormData>({
+      query: (formData) => ({
+        url: apiUrls.uploadListingImages,
+        method: "POST",
+        body: formData,
+      }),
+      invalidatesTags: ["CURRENT_LISTING"],
```
Comment (o-bilyi) [inline src/store/api/endpoints/listings/uploadListingImages.ts]: same as above

File: src/store/index.ts [hunk 19]
```diff
@@ -4,15 +4,13 @@ import {
...
+
 import userDateSelectionReducer from "./slices/userDateSelectionSlice";
-import { api } from "./api/reducer";
 
 export const store = configureStore({
   reducer: {
     [api.reducerPath]: api.reducer,
-    // [authSlice.name]: authSlice.reducer,
-    listingsInfo: listingsInfoReducer,
     userDateSelection: userDateSelectionReducer,
```
Comment (o-bilyi) [inline src/store/index.ts:14]: are you sure you need useDateSelection?

File: src/store/api/lib/constants.ts [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+  getVerifiedListings: "/listings/verified",
+  getListingsCategories: "/listings/categories",
+  getListingsTypeOfPlace: "/listings/typeofplace",
+  requestToDeleteListing: "/listings/listing/delete",
+  requestToCreateListing: "/listings/listing/create",
+
+  uploadListingImages: "/listings/images/upload",
+};
+
+export const ApiTagsTypes = [
```
Comment (o-bilyi) [inline src/store/api/lib/constants.ts:26]: create const for every tag name and reuse theme in your invalidatesTags

**Review by o-bilyi — CHANGES_REQUESTED** [review]
left comments, fix please before continue working

**Comment by oleksandrkovaliuk** [comment]
> left comments, fix please before continue working

All fixes are good to go

**Review by o-bilyi — APPROVED** [review]

**Pull request merged**